internal/api/v1/applications.go

```diff
@@ -81,7 +81,52 @@ func (hc ApplicationsController) Create(w http.ResponseWriter, r *http.Request)
 	return nil
 }
 
-// Index handles the API endpoint GET /orgs/:org/applications
+// Index handles the API endpoint GET /applications
+// It lists all the known applications, with and without workload.
+func (hc ApplicationsController) FullIndex(w http.ResponseWriter, r *http.Request) APIErrors {
+	ctx := r.Context()
+
+	cluster, err := kubernetes.GetCluster(ctx)
+	if err != nil {
+		return InternalError(err)
+	}
+
+	// naive: list namespaces, then all apps in each namespace ...
+	// can we query kube for all apps directly, across namespaces ...
+	// needs label selector, pods, and app CRD resources (workloads, and undeployed apps).
+	// the naive way seems to be much easier to implement right now.
```

Comment:
Looking at how `application.ListApps` is implemented, it should be relatively easy to get a full list if we just passed an empty namespace name ([here](https://github.com/epinio/epinio/blob/a67014c320c915ad76986414a13b4d4c537ca59d/internal/application/application.go#L114) and [here](https://github.com/epinio/epinio/blob/a67014c320c915ad76986414a13b4d4c537ca59d/internal/application/application.go#L160)).

I'm fine keeping this implementation for now but we should avoid N+1 queries as much as we can. With lots of namespaces and lots of apps, this could be an issue.

Reply:
If a simple solution like this is possible, we could even use just a single endpoint both for `AllApps` and `Apps`. After all, the namespace can be seen as a filter on the `Apps` endpoint.

Reply:
Heh. ... I would like to defer and create a new ticket for this. already deeper in the work for #802.

Reply:
Ticket is #805